modules/cats/src/main/scala/evo/derivation/cats/EvoEq.scala

```diff
@@ -0,0 +1,94 @@
+package evo.derivation.cats
+
+import evo.derivation.LazySummon.LazySummonByConfig
+import evo.derivation.{LazySummon, ValueClass}
+import evo.derivation.config.Config
+import evo.derivation.internal.{Matching, mirroredNames, underiveableError}
+
+import scala.compiletime.{summonFrom, summonInline}
+import scala.deriving.Mirror
+
+trait Equ[A]:
```

Comment:
What is the motivation behind introducing both `Equ` and `EvoEq`?

Reply:
Equ is an intermediate trait, probably it would be wise to move inside EvoEq object. 
Equ is evidence there is one of the `cats.Eq` or `scala.Equals` instances.
I've introduced `Equ` after I've found there no `Eq` instances for `java.time` things